### PR TITLE
VUE-490 Add gzip to dev server

### PR DESCRIPTION
**Pull request by ryan-ludwig**

### CODE REVIEW

**Comment by ryan-ludwig** [comment]
Cool to see the lighthouse perf diff: https://googlechrome.github.io/lighthouse-ci/viewer/?baseReport=https%3A%2F%2Fstorage.googleapis.com%2Flighthouse-infrastructure.appspot.com%2Freports%2F1614040055719-50033.report.html&compareReport=https%3A%2F%2Fstorage.googleapis.com%2Flighthouse-infrastructure.appspot.com%2Freports%2F1614040548932-98785.report.html

I think I might have messed up the package-lock though which is causing the tests to fail? Any ideas on that?

**Comment by ryan-ludwig** [comment]
> Cool to see the lighthouse perf diff: https://googlechrome.github.io/lighthouse-ci/viewer/?baseReport=https%3A%2F%2Fstorage.googleapis.com%2Flighthouse-infrastructure.appspot.com%2Freports%2F1614040055719-50033.report.html&compareReport=https%3A%2F%2Fstorage.googleapis.com%2Flighthouse-infrastructure.appspot.com%2Freports%2F1614040548932-98785.report.html
> 
> I think I might have messed up the package-lock though which is causing the tests to fail? Any ideas on that?

Wrong Node versions maybe?

**Comment by emuvente** [comment]
The error in the build is `git@github.com: Permission denied (publickey).` from trying to fetch `ssh://git@github.com/kiva/flag-icon-css.git`. I don't know why that wouldn't be working for this pr though.

It looks like there are massive changes to the package lock, did you install while running npm 7? I don't think we've done that before, so those changes are probably the migrating of the lockfile to the v2 format (I'm pretty sure it's backwards compatible though, so I wouldn't expect that to be the cause of any problem)

File: server/index.js [hunk 1]
```diff
@@ -33,6 +34,9 @@ const cache = initCache(config.server);
 const app = express();
 const port = argv.port || config.server.port;
 
+// gzip
+app.use(compression());
```
Comment (mcstover) [inline server/index.js:38]: This needs to be gated behind a config that is only set to `true` when starting in local mode.

When running on actual servers gzip is activated at a different layer.

**Review by mcstover — APPROVED** [review]

**Review by emuvente — APPROVED** [review]
👍

**Pull request merged**